### PR TITLE
Add auth header with github token for reading licenses

### PR DESCRIPTION
**Pull request by abhay-krishna**
Passing auth token in header so that rate limiting doesn't occur.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: generate-attribution/generate-attribution-file.js [hunk 1]
```diff
@@ -338,6 +340,15 @@ async function populateVersionAndModuleFromDep(dependencies) {
     return finalDeps;
 }
 
+function generateAuthorizationHeader() {
+    const githubToken = await fsPromises.readFile("/secrets/github-secrets/token", 'utf8');
```
Comment (jaxesn) [inline generate-attribution/generate-attribution-file.js:344]: Probably not the most elegant but just add a .stat call before hand to make sure the file exists. If it doesn't, just return an empty object. I think ideally wed pass this in as a cli arg or use an env var, but this is totally fine

File: generate-attribution/generate-attribution-file.js [hunk 2]
```diff
@@ -338,6 +340,23 @@ async function populateVersionAndModuleFromDep(dependencies) {
     return finalDeps;
 }
 
+function generateAuthorizationHeader() {
+    const githubTokenFile = "/secrets/github-secrets/token";
+    fs.stat(githubTokenFile, (error, stats) => {
```
Comment (jaxesn) [inline generate-attribution/generate-attribution-file.js:345]: Sorry this is the pattern to follow, in the catch block return empty object, that means file doesn't exist

https://github.com/aws/eks-distro-build-tooling/blob/ac2191b893bfde5628e758007c7fda5306cd7e36/generate-attribution/generate-attribution-file.js#L220

Comment (abhay-krishna) [reply generate-attribution/generate-attribution-file.js]: Thanks @jaxesn, updated the PR accordingly.

File: generate-attribution/generate-attribution-file.js [hunk 3]
```diff
@@ -338,6 +340,21 @@ async function populateVersionAndModuleFromDep(dependencies) {
     return finalDeps;
 }
 
+async function generateAuthorizationHeader() {
+    const githubTokenFile = "/secrets/github-secrets/token";
+    try {
+        await fsPromises.access(githubTokenFile);
+        const githubToken = await fsPromises.readFile(possiblePath, 'utf8');
```
Comment (jaxesn) [inline generate-attribution/generate-attribution-file.js:347]:
```suggestion
        const githubToken = await fsPromises.readFile(githubTokenFile, 'utf8');
```

**Review by jaxesn — APPROVED** [review]
/lgtm

**Pull request merged**